pyeapi/api/vrfs.py

```diff
@@ -0,0 +1,358 @@
+                if 0 <= local_assignment <= 4294967295:
+                    return True
+    except ValueError:
+        pass
+    return False
+
+
+class Vrfs(EntityCollection):
+    """The Vrfs class provides a configuration resource for VRFs
+
+    The Vrfs class is derived from ResourceBase a standard set of methods
+    for working with VRF configurations on an EOS node.
+
+    """
+
+    def get(self, value):
+        """Returns the VRF configuration as a resource dict.
+
+        Args:
+            value (string): The vrf name to retrieve from the
+                running configuration.
+
+        Returns:
+            A Python dict object containing the VRF attributes as
+                key/value pairs.
+
+        """
+        config = self.get_block('vrf definition %s' % value)
+        if not config:
+            return None
+        response = dict(vrf_name=value)
+        response.update(self._parse_rd(config))
+        response.update(self._parse_description(config))
+        config = self.get_block('no ip routing vrf %s' % value)
+        if config:
+            response['ipv4routing'] = False
+        else:
+            response['ipv4routing'] = True
+        config = self.get_block('no ipv6 unicast-routing vrf %s' % value)
+        if config:
+            response['ipv6routing'] = False
+        else:
+            response['ipv6routing'] = True
+
+        return response
+
+    def _parse_rd(self, config):
+        """ _parse_rd scans the provided configuration block and extracts
+        the vrf rd. The return dict is intended to be merged into the response
+        dict.
+
+        Args:
+            config (str): The vrf configuration block from the nodes running
+                configuration
+
+        Returns:
+            dict: resource dict attribute
+        """
+        match = RD_RE.search(config)
+        if match:
+            value = match.group('value')
+        else:
+            value = match
+        return dict(rd=value)
+
+    def _parse_description(self, config):
+        """ _parse_description scans the provided configuration block and
+        extracts the vrf description value. The return dict is intended to
+        be merged into the response dict.
+
+        Args:
+            config (str): The vrf configuration block from the nodes
+                running configuration
+
+        Returns:
+            dict: resource dict attribute
+        """
+        value = DESCRIPTION_RE.search(config).group('value')
+        return dict(description=value)
+
+    def getall(self):
+        """Returns a dict object of all VRFs in the running-config
+
+        Returns:
+            A dict object of VRF attributes
+
+        """
+        vrfs_re = re.compile(r'(?<=^vrf definition\s)(\w+)', re.M)
+
+        response = dict()
+        for vrf in vrfs_re.findall(self.config):
+            response[vrf] = self.get(vrf)
+        return response
+
+    def create(self, vrf_name):
```

Comment:
This one is a corner case because we generally use `create` to get the element instantiated and into a base state, then use the helper functions to set the numerous attrs.  Since vrfs are so simple, it might make sense to have the rd passed in on create as a convenience.